addon_service/management/commands/fix_migrating_users_with_mutiple_external_accounts.py

```diff
@@ -0,0 +1,271 @@
+        check_fields(external_account, ["oauth_secret"])
+        credentials = AccessTokenCredentials(access_token=external_account.oauth_secret)
+    elif external_service.wb_key == "dropbox" and not external_account.refresh_token:
+        check_fields(external_account, ["oauth_key"])
+        credentials = AccessTokenCredentials(access_token=external_account.oauth_key)
+    else:
+        check_fields(external_account, ["oauth_key"])
+        credentials = AccessTokenCredentials(access_token=external_account.oauth_key)
+    return credentials
+
+
+# Ported over from migrate_authorized_account.py
+def check_fields(external_account: ExternalAccount, fields: list[str]):
+    errors = []
+    for field in fields:
+        if getattr(external_account, field, None) is None:
+            error_string = f"Required field <<{field}>> is None"
+            errors.append(error_string)
+    if errors:
+        raise CredentialException(errors)
+
+
+# Ported over from migrate_authorized_account.py
+def get_node_guid(id_):
+    content_type_id = cache.get_or_set(
+        "node_contenttype_id",
+        lambda: ContentType.objects.using("osf")
+        .get(app_label="osf", model="abstractnode")
+        .id,
+        timeout=None,
+    )
+    return (
+        Guid.objects.filter(content_type_id=content_type_id, object_id=id_).first()._id
+    )
+
+
+# return a list of tuples of (user_guid, provider_name_with_duplicate_accounts)
+def get_target_user_guids_and_provider():
+    return (
+        ExternalAccount.objects.values("osfuser__guids___id", "provider")
+        .annotate(account_per_provider=Count("id"))
+        .filter(account_per_provider__gt=1)
+        .values_list("osfuser__guids___id", "provider")
+    )
+
+
+def get_node_settings_for_user_and_provider(user_guid, provider):
+    UserSettings = provider_to_user_settings[provider]
+    user_setting = UserSettings.objects.get(
+        owner=OsfUser.objects.get(guids___id=user_guid)
+    )
+    return getattr(user_setting, f"{provider}nodesettings_set").all()
+
+
+def get_configured_addon(node_guid, provider):
+    resource_reference = ResourceReference.objects.get_or_create(
+        resource_uri=f"{OSF_BASE}/{node_guid}"
+    )[0]
+    if provider in storage_provider:
+        ConfiguredAddon = ConfiguredStorageAddon
+    elif provider in citation_provider:
+        ConfiguredAddon = ConfiguredCitationAddon
+
+    try:
+        configured_addon = ConfiguredAddon.objects.get(
+            authorized_resource=resource_reference
+        )
+    except ConfiguredAddon.DoesNotExist:
+        print("ConfiguredAddon not found")
+
+    return configured_addon
+
+
+def get_external_service(provider):
+    return ExternalService.objects.get(wb_key=provider)
+
+
+def get_user_reference(user_guid):
+    return UserReference.objects.get_or_create(user_uri=f"{OSF_BASE}/{user_guid}")
+
+
+def get_or_create_authorized_account(external_account, provider, user_guid):
```

Comment:
Could you please add some logging in this function to say what actions it's performing, e.g. `Creating AuthorizedAccount on {provider} for {user}`, `Found AuthorizedAccount {id} for {user}`, `Found shared credentials on {provider} for users {users_0...n}`, etc?